FullStackMVCAPP/FullStackMVCAPP/Controllers/CastleController.cs

```diff
@@ -12,11 +15,12 @@ public class CastleController : Controller
         // GET: Castle
         public ActionResult Index()
         {
-            using (var db = new GOTContext()) 
-            {
-                var castleList = db.Castles.ToList();
-                return View(castleList);
-            }
+            GOTContext  context = new DataContext.GOTContext();
+            CastleRepository castleRepository = new CastleRepository();
```

Comment:
Creating a new repository class and passing that to the service shouldn't be required in the controller, this should be removed. The service should handle this within that class and it appears to do so looking at that code.
Likewise for the context, this is created in the service so isn't needed here.